examples/vision/keypoint_detection.py

```diff
@@ -0,0 +1,440 @@
+        if self.train:
+            np.random.shuffle(self.indexes)
+
+    def __getitem__(self, index):
+        indexes = self.indexes[index * self.batch_size : (index + 1) * self.batch_size]
+        image_keys_temp = [self.image_keys[k] for k in indexes]
+        (images, keypoints) = self.__data_generation(image_keys_temp)
+
+        return (images, keypoints)
+
+    def __data_generation(self, image_keys_temp):
+        batch_images = np.empty((self.batch_size, IMG_SIZE, IMG_SIZE, 3), dtype="int")
+        batch_keypoints = np.empty(
+            (self.batch_size, 1, 1, NB_KEYPOINTS), dtype="float32"
+        )
+
+        for i, key in enumerate(image_keys_temp):
+            data = get_dog(key)
+            current_keypoint = np.array(data["joints"])[:, :2]
+            kps = []
+
+            # To apply our data augmentation pipeline, we first need to
+            # form Keypoint objects with the original coordinates.
+            for j in range(0, len(current_keypoint)):
+                kps.append(Keypoint(x=current_keypoint[j][0], y=current_keypoint[j][1]))
+
+            # We then project the original image and its keypoint coordinates.
+            current_image = data["img_data"]
+            kps_obj = KeypointsOnImage(kps, shape=current_image.shape)
+
+            # Apply the augmentation pipeline.
+            (new_image, new_kps_obj) = self.aug(image=current_image, keypoints=kps_obj)
+
+            # If the image is RGBA convert it to RGB.
+            if new_image.shape[-1] == 4:
+                new_image = Image.fromarray(new_image)
+                new_image = np.array(new_image.convert("RGB"))
+            batch_images[i,] = new_image
+
+            # Parse the coordinates from the new keypoint object.
+            kpTemp = []
+            for keypoint in new_kps_obj:
+                kpTemp.append(np.nan_to_num(keypoint.x))
+                kpTemp.append(np.nan_to_num(keypoint.y))
+
+            # More on why this reshaping later.
+            batch_keypoints[i,] = np.array(kpTemp).reshape(1, 1, 24 * 2)
+
+        # Scale the coordinates to [0, 1] range.
+        batch_keypoints = batch_keypoints / IMG_SIZE
+
+        return (batch_images, batch_keypoints)
+
+
+"""
+To know more about how to operate with keypoints in `imgaug` check out [this document](https://imgaug.readthedocs.io/en/latest/source/examples_keypoints.html). 
+"""
+
+"""
+## Define augmentation transforms
+"""
+
+train_aug = iaa.Sequential(
+    [
+        iaa.Resize(IMG_SIZE, interpolation="linear"),
+        iaa.Fliplr(0.3),
+        # `Sometimes()` applies a function randomly to the inputs with
+        # a given probability (0.3, in this case).
+        iaa.Sometimes(0.3, iaa.Affine(rotate=10, scale=(0.5, 0.7))),
+    ]
+)
+
+test_aug = iaa.Sequential([iaa.Resize(IMG_SIZE, interpolation="linear")])
+
+"""
+## Create training and validation splits
+"""
+
+train_keys, validation_keys = train_test_split(samples, test_size=0.15, random_state=42)
```

Comment:
Prefer doing this manually, we don't need to add a dependency on sklearn just for this